c/tskit/trees.c

```diff
@@ -2558,14 +2738,46 @@ sample_sets_to_bit_array(const tsk_treeseq_t *self, const tsk_size_t *sample_set
     return ret;
 }
 
+static int
+check_sites(const tsk_id_t *sites, tsk_size_t num_sites, tsk_size_t num_site_rows)
+{
+    int ret = 0;
+    tsk_id_t id;
+    tsk_size_t i;
+
+    if (sites == NULL) {
+        goto out;
+    }
+
+    for (i = 0; i < num_sites - 1; i++) {
+        id = sites[i];
+        if (id == TSK_NULL) {
+            ret = TSK_ERR_BAD_PARAM_VALUE; // TODO: new err type? TSK_BAD_SITE_VALUE?
+            goto out;
+        }
+        if (id > sites[i + 1]) {
+            ret = TSK_ERR_UNSORTED_SITES; // TODO: this checks no repeats, but error is
+                                          // ambiguous
+            goto out;
+        }
+    }
+    // check if the last or only value is null
+    if (num_sites != 0
```

Comment:
Why are we doing this?

Reply:
We check if every item is non-null and ordered by iterating from 0->n-1. We then need to check the last item if it's null. Also, since we have checked if the sites are ordered, by checking if the last site exceeds the length of the site rows table, we can ensure we're not specifying a site that's out of bounds.